packages/database/src/modelClasses/Entity.js

```diff
@@ -112,24 +111,67 @@ class EntityType extends DatabaseType {
     return parent.type === COUNTRY;
   }
 
+  async getAncestors(hierarchyId, ancestorEntityType, parentOnly) {
+    return this.model.getAncestorsOfEntity(this.id, hierarchyId, ancestorEntityType, parentOnly);
+  }
+
+  async getDescendants(hierarchyId, descendantEntityType, childrenOnly) {
+    return this.model.getDescendantsOfEntity(
+      this.id,
+      hierarchyId,
+      descendantEntityType,
+      childrenOnly,
+    );
+  }
+
+  async getAncestorOfType(entityType, hierarchyId) {
+    if (this.type === entityType) return this;
+    const [ancestor] = await this.getAncestors(hierarchyId, entityType);
```

Comment:
It would be good to align the parameter order between `getAncestorsOfType` and `getAncestors`. Currently:

```js
getAncestors(hierarchyId, ancestorEntityType, parentOnly)
getAncestorOfType(entityType, hierarchyId)
```

I'll take this one step further and suggest that instead of supporting both methods, we could expose a single method:

```js
getAncestors(hierarchyId, conditions)
// or, according to our needs
getAncestors(hierarchyId, options)
```
Where `type` can be optionally specified in the 2nd parameter


Similarly for their descendant counterparts

Reply:
The alignment of parameter order is implemented in part 3 🎉 

For the second step of the suggestion, I'll think on that, but not implement it in this PR due to the conflicts it would create with part 3. Feel free to raise it again there if you feel strongly, and I may implement it there either way

Reply:
For the second part: feel free to not change it in this PR!

If I feel strongly about it or not, well I would definitely attempt it and see how the code looks like (expecting it to look better) but it's not the end of the world if you are not in favour of it. Perhaps it's a bit of a personal style choice